rpcs3/Emu/Cell/PPUInterpreter.cpp

```diff
@@ -358,6 +358,29 @@ class ppu_scale_table_t
 }
 const g_ppu_scale_table;
 
+constexpr u32 ppu_nan_u32 = 0x7FC00000u;
+const f32 ppu_nan_f32 = *reinterpret_cast<const f32*>(&ppu_nan_u32);
+
+__m128 vec_handle_nan(__m128 val)
+{
+	const auto mask_nan = _mm_cmpunord_ps(val, val);
+	const auto mask_not_nan = _mm_cmpord_ps(val, val);
+
+	const auto nans = _mm_castsi128_ps(_mm_set1_epi32(ppu_nan_u32));
+
+	const auto nan_part = _mm_and_ps(mask_nan, nans);
+	const auto not_nan_part = _mm_and_ps(mask_not_nan, val);
```

Comment:
vector select can be optimized into `result = (nans & is_nan) | (val & ~is_nan);`
try to use use v128 as well as it has builtin operator for these.
nans can be constructed as v128 using v128::from32p(ppu_man_f32).

Reply:
nice, thx